File: python3/1833. Maximum Ice Cream Bars.py
'''

'''

class Solution:
    def maxIceCream(self, costs: List[int], coins: int) -> int:
        costs.sort()
        num = 0
        for i in range(len(costs)):
            if coins > 0 and coins >= costs[i]:
                coins -= costs[i] 
                num += 1
        return num

File: cpp/1833_Maximum_Ice_Cream_Bars.cpp
/*

*/

class Solution {
public:
    int maxIceCream(vector<int>& costs, int coins) {
        sort(costs.begin(), costs.end());
        int num=0;
        for(int i=0; i < costs.size(); i++){
            if(coins < costs[i])
                break;
            coins -= costs[i];
            num++;
        }
        return num;
    }
};
